src/main/java/interface_adapter/ViewModel.java

```diff
@@ -4,4 +4,17 @@
 
 public abstract class ViewModel {
 
+    private String viewName;
+
+    public ViewModel(String viewName) {
+        this.viewName = viewName;
+    }
+    public String getViewName() {
+        return this.viewName;
+    }
+
+    public abstract void firePropertyChanged();
```

Comment:
I don't understand what's going on here and why the method is abstract, unless its just a filler to be implemented later